src/dbnode/storage/cleanup.go

```diff
@@ -301,26 +297,75 @@ func (m *cleanupManager) commitLogTimes(t time.Time) (time.Time, []time.Time, er
 	// are only retained for a period of 1-2 days (at most), after we which we'd live we with the
 	// data loss.
 
-	candidateTimes := timesInRange(earliest, latest, blockSize)
+	files, err := m.commitLogFilesFn(m.opts.CommitLogOptions())
+	if err != nil {
+		return nil, err
+	}
 	namespaces, err := m.database.GetOwnedNamespaces()
 	if err != nil {
-		return time.Time{}, nil, err
+		return nil, err
 	}
-	cleanupTimes := filterTimes(candidateTimes, func(t time.Time) bool {
+
+	var outerErr error
+	filesToCleanup := filterCommitLogFiles(files, func(start time.Time, duration time.Duration) bool {
+		if outerErr != nil {
+			return false
+		}
```

Comment:
Is this a necessary skip?  It seems like we'll reach same result whether we skip entries after we've seen a single entry.  Just because if so, then we don't have to add test coverage for these lines if it doesn't make a difference.

Reply:
Yeah I guess its just the difference between first error wins / last error wins. Some of the operations are expensive in that they have to go to disk so returning early is nice in that sense too, but optimizing for the error case is kind of dumb. I'll remove it.